stripped/colabs/GrupowanieEM_clean_output_stripped.py

```diff
@@ -0,0 +1,507 @@
+
+
+NASA2020_annual_anotated = NASA2020_annual_anotated.merge( NASA2020_rainfall_sum, how='inner', on=['lon', 'lat'])
+
+NASA2020_annual_anotated.head()
+
+def assert_add_column(dataset: pd.DataFrame, feature_name: str, month: int) -> None:
+  assert (1<= month <= 12), f'Invalid month number'
+  assert feature_name in dataset.columns, f'The attribute "{feature_name}" is not a column in the DataFrame (full_dataset)'
+
+
+#full_dataset - dataframe with monthly data spanning over a (particular) year
+def add_column_to_dataframe(df: pd.DataFrame, full_dataset: pd.DataFrame, feature_name: str, month: int) -> pd.DataFrame:
+  # function adds a column to the given dataframe, containing monthly data for the specified feature (feature_name) and month
+  assert_add_column(full_dataset, feature_name, month)
+
+  feature_df = full_dataset[['lon', 'lat', feature_name, 'Month']]
+  feature_df_month = feature_df[feature_df['Month'] == month]
+  feature_df_month = feature_df_month.drop(columns=['Month']).rename(columns = {feature_name: feature_name+str(month)})
+
+  return df.merge(feature_df_month, how='inner', on=['lon', 'lat'])
+
+
+#adding columns with monthly data for selected features and months
+selected_months = [month for month in range(5,10)]
+features_names = ['PotEvap', 'Evap', 'Rainf', 'Albedo']
+
+for feature in features_names:
+   for month in selected_months:
+       NASA2020_annual_anotated = add_column_to_dataframe(NASA2020_annual_anotated, NASA_2020_an_pd, feature, month)
+
+selected_months2 = [month for month in range(3,11)]
+
+for month in selected_months2:
+       NASA2020_annual_anotated = add_column_to_dataframe(NASA2020_annual_anotated, NASA_2020_an_pd, 'RootMoist', month)
+
+
+#adding all columns with "GVEG" monthly data
+year = [month for month in range(1,13)]
+
+for month in year:
+       NASA2020_annual_anotated = add_column_to_dataframe(NASA2020_annual_anotated, NASA_2020_an_pd, 'GVEG', month)
+
+NASA2020_annual_anotated.head(5)
+
+
+NASA2020_annual_anotated_ = NASA2020_annual_anotated.drop(columns=['lon', 'lat'])
+
+#data standarization:
+scaler = preprocessing.StandardScaler()
+standarized = scaler.fit_transform(NASA2020_annual_anotated_)
+
+NASA2020_annual_an_st = pd.DataFrame(standarized, columns=NASA2020_annual_anotated_.columns)
+NASA2020_annual_an_st.head()
+
+#dataframe with labels:
+labels_an = NASA_2020_an_pd[['lon', 'lat', 'pustynia', 'step']].drop_duplicates()
+#desert lables:
+desert_label_an = labels_an['pustynia'].tolist()
+
+#Performing Gaussian Mixture algorithm:
+gm = GaussianMixture(n_components = 2, n_init = 300, max_iter=200, init_params= 'k-means++', covariance_type='spherical', random_state=42)
+gm_result = gm.fit_predict(NASA2020_annual_an_st)
+
+metrics = {
+'accuracy' : accuracy_score(gm_result, desert_label_an),
+'precision' : precision_score(gm_result, desert_label_an),
+'jaccard' : jaccard_score(gm_result, desert_label_an),
+}
+print(metrics)
```

Comment:
Tak też moze byc, ale jeśli by zostało:
```suggestion
{
'accuracy' : accuracy_score(gm_result, desert_label_an),
'precision' : precision_score(gm_result, desert_label_an),
'jaccard' : jaccard_score(gm_result, desert_label_an),
}
```
To też by się wyświetliło